compiler/code-generator.cpp

```diff
@@ -1619,6 +1619,7 @@ CodeGenerator::EmitForStmt(ForStmt* stmt)
     auto body = stmt->body();
     bool body_always_exits = body->flow_type() == Flow_Return ||
                              body->flow_type() == Flow_Break;
+    body_always_exits = body_always_exits && !(stmt->has_continue());
```

Comment:
This would read better as:

    bool body_always_exits = false;
    if (body->flow_type() == Flow_Return || body->flow_type() == Flow_Break) {
        if (!stmt->has_continue())
            body_always_exits = true;
    }